tests/load/filesystem/test_filesystem_client.py

```diff
@@ -421,3 +421,19 @@ def test_get_storage_version_invalid(invalid_version_info: Union[str, Dict[str,
     else:
         with pytest.raises(UnsupportedStorageVersionException):
             client.get_storage_versions()
+
+
+def test_list_dlt_table_files_with_separator_in_pipeline_name() -> None:
```

Comment:
Very good, let's cover more test cases with `pytest.mark.parametrize`, see the example test above: e.g. single `__`, multiple `__`, consider other edge cases.